### DISCUSSION TITLE
About new user Auth with OTP

### DISCUSSION THREAD

**Discussion by web5173** [pocketbase/pocketbase#5661] [Q&A]
`// send OTP email to the provided auth record
const req = await pb.collection('users').requestOTP('test@example.com');`

`[0.00ms] SELECT `users`.* FROM `users` WHERE `email`='test@example.com' LIMIT 1
ERROR POST /api/collections/u/request-otp
└─ failed to fetch u record with email test@example.com: sql: no rows in result set`

When a new user requests an OTP, the server gets an error and the user does not receive the OTP. Can I change the logic to automatically create this user when it doesn't exist?

**Comment by ganigeorgiev** [comment]
No, at least not directly. The `OnRecordRequestOTPRequest` also only fires if a matching auth record is found.

I'll have to think a little more on it and may reconsider before the final v0.23 to trigger `OnRecordRequest*` hooks even if they fail to find a matching record but for now I recommend to send a separate create request and if it fails you can assume that the record exist and send a following OTP request.

**Comment by web5173** [comment]
问题解决，非常感谢！期盼v0.23
